starboard/contrib/cast/cast_starboard_api/samples/BUILD.gn

```diff
@@ -0,0 +1,51 @@
+# Copyright 2023 The Cobalt Authors. All rights reserved.
+import("//starboard/build/config/starboard_target_type.gni")
+
+group("cast") {
+  if (build_with_separate_cobalt_toolchain) {
+    public_deps = [ ":cast_starboard_api($starboard_toolchain)" ]
+  }
+}
+
+if (build_with_separate_cobalt_toolchain) {
+  if (current_toolchain == starboard_toolchain) {
+    starboard_platform_target("cast_starboard_api") {
+      ldflags = [
+        # Hide unwanted symbols, which also shrinks the resulting binary.
+        "-Wl,--version-script=" +
```

Comment:
Are these flags that could potentially go into the existing starboard_platform_group?

Reply:
> Are these flags that could potentially go into the existing starboard_platform_group?

I'm not sure how this would affect the default starboard_platform_group; limiting the externs like this actually reduces binary size by around ~10MB, so I don't feel comfortable doing it to the default lib.

Reply:
I believe we already do something very similar for Evergreen, @y4vor could comment

Reply:
I think @andrewsavage1 is out for next couple days - @niranjanyardi and @y4vor for a quick look. The rest of the PR seems to have everything addressed and ready to land.

Reply:
Resolving since @y4vor has approved. If these flags prove useful and stable for libstarboard_platform_group.so, they can be refactored in a separate change.